flash_stm32h7x.c fails to build

`./scripts/twister -T samples/drivers/flash_shell/ -p nucleo_h743zi`

gets:
```
[ 35%] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/flash/flash_stm32h7x.c.obj
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:608:3: error: 'struct flash_stm32_priv' has no member named 'pclken'
  608 |  .pclken = { .bus = STM32_CLOCK_BUS_AHB3,
      |   ^~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:608:15: error: 'struct k_sem' has no member named 'bus'
  608 |  .pclken = { .bus = STM32_CLOCK_BUS_AHB3,
      |               ^~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:608:21: error: 'STM32_CLOCK_BUS_AHB3' undeclared here (not in a function)
  608 |  .pclken = { .bus = STM32_CLOCK_BUS_AHB3,
      |                     ^~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:609:8: error: 'struct k_sem' has no member named 'enr'
  609 |       .enr = LL_AHB3_GRP1_PERIPH_FLASH },
      |        ^~~
In file included from /home/galak/git/zephyr/include/toolchain/gcc.h:69,
                 from /home/galak/git/zephyr/include/toolchain.h:43,
                 from /home/galak/git/zephyr/include/kernel_includes.h:19,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:10:
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c: In function 'stm32h7_flash_init':
/home/galak/git/zephyr/include/device.h:79:39: error: '__device_dts_ord_STM32_CLOCK_CONTROL_NODE_ORD' undeclared (first use in this function)
   79 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/galak/git/zephyr/include/toolchain/common.h:124:26: note: in definition of macro '_DO_CONCAT'
  124 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/galak/git/zephyr/include/device.h:79:31: note: in expansion of macro '_CONCAT'
   79 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/home/galak/git/zephyr/include/device.h:256:37: note: in expansion of macro 'DEVICE_NAME_GET'
  256 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:270:34: note: in expansion of macro 'DEVICE_DT_NAME_GE'
  270 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:625:29: note: in expansion of macro 'DEVICE_DT_GET'
  625 |  const struct device *clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
      |                             ^~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:79:39: note: each undeclared identifier is reported only once for each function it appears in
   79 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/galak/git/zephyr/include/toolchain/common.h:124:26: note: in definition of macro '_DO_CONCAT'
  124 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/galak/git/zephyr/include/device.h:79:31: note: in expansion of macro '_CONCAT'
   79 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/home/galak/git/zephyr/include/device.h:256:37: note: in expansion of macro 'DEVICE_NAME_GET'
  256 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:270:34: note: in expansion of macro 'DEVICE_DT_NAME_GE'
  270 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:625:29: note: in expansion of macro 'DEVICE_DT_GET'
  625 |  const struct device *clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
      |                             ^~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:628:6: error: implicit declaration of function 'clock_control_on' [-Werror=implicit-function-declaration]
  628 |  if (clock_control_on(clk, (clock_control_subsys_t *)&p->pclken) != 0) {
      |      ^~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:628:29: error: 'clock_control_subsys_t' undeclared (first use in this function)
  628 |  if (clock_control_on(clk, (clock_control_subsys_t *)&p->pclken) != 0) {
      |                             ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:628:53: error: expected expression before ')' token
  628 |  if (clock_control_on(clk, (clock_control_subsys_t *)&p->pclken) != 0) {
      |                                                     ^
/home/galak/git/zephyr/drivers/flash/flash_stm32h7x.c:624:27: error: unused variable 'p' [-Werror=unused-variable]
  624 |  struct flash_stm32_priv *p = FLASH_STM32_PRIV(dev);
      |                           ^
cc1: all warnings being treated as errors

```

--- a/drivers/flash/flash_stm32.h
+++ b/drivers/flash/flash_stm32.h
@@ -8,14 +8,16 @@
 #ifndef ZEPHYR_DRIVERS_FLASH_FLASH_STM32_H_
 #define ZEPHYR_DRIVERS_FLASH_FLASH_STM32_H_
 
-#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_flash_controller), clocks)
+#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_flash_controller), clocks) || \
+	DT_NODE_HAS_PROP(DT_INST(0, st_stm32h7_flash_controller), clocks)
 #include <drivers/clock_control.h>
 #include <drivers/clock_control/stm32_clock_control.h>
 #endif
 
 struct flash_stm32_priv {
 	FLASH_TypeDef *regs;
-#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_flash_controller), clocks)
+#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_flash_controller), clocks) || \
+	DT_NODE_HAS_PROP(DT_INST(0, st_stm32h7_flash_controller), clocks)
 	/* clock subsystem driving this peripheral */
 	struct stm32_pclken pclken;
 #endif

--- a/drivers/flash/flash_stm32h7x.c
+++ b/drivers/flash/flash_stm32h7x.c
@@ -605,8 +605,8 @@ void flash_stm32_page_layout(const struct device *dev,
 
 static struct flash_stm32_priv flash_data = {
 	.regs = (FLASH_TypeDef *) DT_INST_REG_ADDR(0),
-	.pclken = { .bus = STM32_CLOCK_BUS_AHB3,
-		    .enr = LL_AHB3_GRP1_PERIPH_FLASH },
+	.pclken = { .bus = DT_INST_CLOCKS_CELL(0, bus),
+		    .enr = DT_INST_CLOCKS_CELL(0, bits)},
 };
 
 static const struct flash_driver_api flash_stm32h7_api = {

--- a/dts/arm/st/h7/stm32h7.dtsi
+++ b/dts/arm/st/h7/stm32h7.dtsi
@@ -44,6 +44,7 @@
 			label = "FLASH_CTRL";
 			reg = <0x52002000 0x400>;
 			interrupts = <4 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00000100>;
 
 			#address-cells = <1>;
 			#size-cells = <1>;

